### ISSUE TITLE
@salesforce-ux/slds/enforce-bem-class warns against classes like "slds-dl--horizontal__label"

### ISSUE THREAD

**Issue by aheber** [salesforce-ux/slds-linter#39]
This one is a little complicated it seems. The stylelint side recommended changings `slds-dl_horizontal__label` to `slds-dl--horizontal__label` which generally makes sense.

This eslint plugin makes a complaint about `slds-dl_horizontal__label` not following BEM naming conventions. When I switch it to `slds-dl--horizontal__label` I get the same error, so it seems the plugin isn't happy with either. 

`error  The class 'slds-dl--horizontal__label' does not follow BEM naming convention   @salesforce-ux/slds/enforce-bem-class`

If I look at the Lightning Design System docs it appears to recommend the old style still.
https://www.lightningdesignsystem.com/2e1ef8501/p/62faac-description-list

So now I'm a little confused, why does the recommended change still get flagged, and why does the tool's recommended change not conform to the documentation? If I look at the compiled SLDS sheet that is running in my Salesforce org it allows for either name.

```css
    .slds-scope .slds-dl_horizontal__label, .slds-scope .slds-dl--horizontal__label {
        width: 30%;
        padding-right: var(--lwc-spacingSmall,0.75rem)
    }
```

Please help me understand the correct behavior and if necessary, push the doc owners to update their content.

**Comment by kenamverma** [comment]
Hi @aheber, thanks for letting us know this. We are moving away from BEM and the [documentation](https://www.lightningdesignsystem.com/2e1ef8501/p/62faac-description-list) is correct. We have corrected the BEM logic and it should be correctly letting you know the violations. Mostly, double dash should be replaced with underscore for a valid className.
Also, the stylelint BEM rule is disabled and currently we have enabled BEM rule only for Eslint configurations.

**Comment by kenamverma** [comment]
We're closing this issue as it's been open for a while without recent activity. If you still have questions or need any help, please don’t hesitate to reach out or open a new issue. Thank you.
